rtt/internal/ChannelBufferElement.hpp

```diff
@@ -129,24 +133,31 @@ namespace RTT { namespace internal {
          */
         virtual void clear()
         {
-	    if(last_sample_p)
-		buffer->Release(last_sample_p);
-	    last_sample_p = 0;
+            if(last_sample_p)
+                buffer->Release(last_sample_p);
+            last_sample_p = 0;
             buffer->clear();
             base::ChannelElement<T>::clear();
         }
 
         virtual bool data_sample(param_t sample)
         {
-            buffer->data_sample(sample);
+            if (!buffer->data_sample(sample)) return false;
             return base::ChannelElement<T>::data_sample(sample);
         }
 
         virtual T data_sample()
         {
             return buffer->data_sample();
         }
-        
+
+        /** Returns a pointer to the ConnPolicy that has been used to construct the underlying buffer.
+        */
+        virtual const ConnPolicy* getConnPolicy() const
```

Comment:
With #114 this method implements [ChannelElementBase::getConnPolicy()](https://github.com/orocos-toolchain/rtt/pull/114/files#diff-51b13774fd6d17869ba8eb2ba6588f7aR238) and the default implementation in [ChannelInterface.cpp:208](https://github.com/orocos-toolchain/rtt/pull/114/files#diff-dff412a478239365fa7700cf3231f045R208) returns a null pointer. Not every channel element stores its connection policy. Only special implementations like data or buffer elements do.

The predecessor of this method was [ChannelElementBase::getConnID()](https://github.com/orocos-toolchain/rtt/pull/114/files#diff-51b13774fd6d17869ba8eb2ba6588f7aL178), which also returned a pointer that could be null.